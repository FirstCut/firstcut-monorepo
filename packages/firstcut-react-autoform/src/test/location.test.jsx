
jest.mock('semantic-ui-react');

import React from 'react';
import { shallow } from 'enzyme';
import { Button } from 'semantic-ui-react';
import renderer from 'react-test-renderer';
import LocationField from '../components/location';

describe('<LocationField />', () => {
  const onChange = jest.fn();
  const locationDisplayName = 'Location Display Name';
  const fieldName = 'Field Name';
  const testRecord = { locationDisplayName };
  const locationComponent = (
    <LocationField
      record={testRecord}
      name={fieldName}
      onChange={onChange}
    />
  );

  const wrapper = shallow(locationComponent);

  test('should contain a clear location button', () => {
    expect(wrapper.find(Button)).toHaveLength(1);
  });

  test('should call onChange with a value of null on clear location', () => {
    wrapper.find(Button).simulate('click');
    expect(onChange).toHaveBeenCalledWith(null, { name: fieldName, value: {} });
  });

  test('should match snapshots', () => {
    const component = renderer.create(locationComponent);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
