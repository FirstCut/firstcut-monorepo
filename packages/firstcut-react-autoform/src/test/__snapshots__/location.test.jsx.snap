// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LocationField /> should match snapshots 1`] = `
<div>
  <input
    control={[Function]}
    name="Field Name"
    onPlaceSelected={[Function]}
    placeholder="Location Display Name"
    types={Array []}
  />
  <button
    attached="bottom"
    onClick={[Function]}
  >
    CLEAR LOCATION
  </button>
</div>
`;
